server/model/monitor.js

```diff
@@ -809,8 +809,39 @@ class Monitor extends BeanModel {
                 bean.important = true;
 
                 if (Monitor.isImportantForNotification(isFirstBeat, previousBeat?.status, bean.status)) {
+                    // Change the notification msg if status becomes UP from DOWN
+                    // Adding msg for the downtime
+                    let downTimeInSeconds;
+                    if (bean.status === UP) {
+                        const downAfterPreviousUpBeat = await R.findOne("heartbeat", " monitor_id = ? AND status = ? AND time > ( SELECT time from heartbeat WHERE monitor_id = ? AND status = ? ORDER BY time DESC LIMIT 1 ) ORDER BY time ASC", [
+                            this.id,
+                            DOWN,
+                            this.id,
+                            UP
+                        ]);
+                        if (downAfterPreviousUpBeat) {
+                            const mostRecentDownBeat = await R.findOne("heartbeat", " monitor_id = ? AND status = ? ORDER BY time DESC", [
+                                this.id,
+                                DOWN
+                            ]);
+                            if (mostRecentDownBeat) {
+                                downTimeInSeconds = await R.getRow("SELECT SUM(duration) AS duration FROM heartbeat WHERE monitor_id = ? AND status = ? AND time > ? OR time = ? AND time < ? OR time = ?", [
+                                    this.id,
+                                    DOWN,
+                                    downAfterPreviousUpBeat.time,
+                                    downAfterPreviousUpBeat.time,
+                                    mostRecentDownBeat.time,
+                                    mostRecentDownBeat.time,
+                                ]);
+                            }
+                        }
+                    }
                     log.debug("monitor", `[${this.name}] sendNotification`);
-                    await Monitor.sendNotification(isFirstBeat, this, bean);
+                    let downTime = 0;
+                    if (downTimeInSeconds) {
+                        downTime = downTimeInSeconds.duration;
+                    }
+                    await Monitor.sendNotification(isFirstBeat, this, bean, downTime);
```

Comment:
```suggestion
                    await Monitor.sendNotification(isFirstBeat, this, bean, downTimeInSeconds.duration);
```

Reply:
What if downTimeInSeconds is undefined here, isn't downTimeInSeconds.duration will give error?